src/core/main.c

```diff
@@ -1253,37 +1253,39 @@ static void bump_file_max_and_nr_open(void) {
 #endif
 
 #if BUMP_PROC_SYS_FS_NR_OPEN
-        int v = INT_MAX;
+        unsigned int v = INT_MAX;
 
-        /* Argh! The kernel enforces maximum and minimum values on the fs.nr_open, but we don't really know
-         * what they are. The expression by which the maximum is determined is dependent on the architecture,
-         * and is something we don't really want to copy to userspace, as it is dependent on implementation
-         * details of the kernel. Since the kernel doesn't expose the maximum value to us, we can only try
-         * and hope. Hence, let's start with INT_MAX, and then keep halving the value until we find one that
-         * works. Ugly? Yes, absolutely, but kernel APIs are kernel APIs, so what do can we do... 🤯 */
+        // cf. https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/fs/file.c?h=v6.8#n27
+	#define BITS_PER_LONG __WORDSIZE
+        #define __const_min(x, y) ((x) < (y) ? (x) : (y))
```

Comment:
`MIN()` macro exists

Reply:
I want to avoid straying from `linux` source as much as possible. I'd prefer to keep this preprocessor expression as-is to avoid divergence.